samples/create-buffers-around-points/src/main/java/com/esri/arcgismaps/sample/createbuffersaroundpoints/screens/CreateBuffersAroundPointsScreen.kt

```diff
@@ -0,0 +1,192 @@
+                        Switch(
+                            checked = shouldUnion,
+                            onCheckedChange = { checked -> mapViewModel.updateUnion(checked) }
+                        )
+                    }
+
+                    Row {
+                        OutlinedButton(onClick = { mapViewModel.clearAll() }) {
+                            Icon(Icons.Filled.Clear, contentDescription = "Clear buffers")
+                            Text(text = " Clear", modifier = Modifier.padding(start = 8.dp))
+                        }
+                    }
+                }
+
+                // Message dialog from the view model
+                mapViewModel.messageDialogVM.apply {
+                    if (dialogStatus) {
+                        MessageDialog(
+                            title = messageTitle,
+                            description = messageDescription,
+                            onDismissRequest = ::dismissDialog
+                        )
+                    }
+                }
+            }
+
+            // Buffer radius input dialog
+            if (isInputVisible) {
+                AlertDialog(
+                    onDismissRequest = { },
+                    title = { Text(text = "Buffer radius (miles)") },
+                    text = {
+                        Column {
+                            Text(text = "Enter a radius between 0 and 300 miles.")
+                            TextField(
+                                value = radiusInput,
+                                onValueChange = { radiusInput = it },
+                                label = { Text("Radius in miles") },
+                                keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number)
+                            )
+                        }
+                    },
+                    confirmButton = {
+                        FilledTonalButton(onClick = {
+                            // Parse the input and submit to the view model
+                            val miles = radiusInput.toDoubleOrNull()
+                            if (miles == null) {
+                                // Show an error dialog via the view model
+                                mapViewModel.messageDialogVM.showMessageDialog(
+                                    "Invalid input",
+                                    "Please enter a valid number"
+                                )
+                            } else {
+                                mapViewModel.submitRadiusMiles(miles)
+                            }
+                        }) {
+                            Text("Done")
+                        }
+                    },
+                    dismissButton = {
+                        Button(onClick = {
+                            // Cancel input
+                            radiusInput = ""
+                            mapViewModel.dismissInputDialog()
+                        }) {
```

Comment:
No need to reset radius here, we can continue using the same input as the previous buffer: 

```suggestion
                        Button(onClick = mapViewModel::dismissInputDialog) {
```